drivers/watchdog/wdt_mcux_rtwdog.c

```diff
@@ -0,0 +1,188 @@
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	RTWDOG_Refresh(base);
+	LOG_DBG("Fed the watchdog");
+
+	return 0;
+}
+
+static void mcux_rtwdog_isr(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	struct mcux_rtwdog_data *data = dev->data;
+	RTWDOG_Type *base = config->base;
+	uint32_t flags;
+
+	flags = RTWDOG_GetStatusFlags(base) & (uint32_t)kRTWDOG_InterruptFlag;
+	RTWDOG_ClearStatusFlags(base, flags);
+
+	if (data->callback) {
+		data->callback(dev, 0);
+	}
+}
+
+static int mcux_rtwdog_init(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+
+	config->irq_config_func(dev);
+
+	return 0;
+}
+
+static DEVICE_API(wdt, mcux_rtwdog_api) = {
+	.setup = mcux_rtwdog_setup,
+	.disable = mcux_rtwdog_disable,
+	.install_timeout = mcux_rtwdog_install_timeout,
+	.feed = mcux_rtwdog_feed,
+};
+
+#define TO_RTWDOG_CLK_SRC(val) _DO_CONCAT(kRTWDOG_ClockSource, val)
+#define TO_RTWDOG_CLK_DIV(val) _DO_CONCAT(kRTWDOG_ClockPrescalerDivide, val)
+
+#define MCUX_RTWDOG_INIT(n)                                                                        \
+                                                                                                   \
+	static struct mcux_rtwdog_data mcux_rtwdog_data_##n;                                       \
+	static void mcux_rtwdog_config_func_##n(const struct device *dev);                         \
+                                                                                                   \
+	static const struct mcux_rtwdog_config mcux_rtwdog_config_##n = {                          \
+		.base = (RTWDOG_Type *)DT_INST_REG_ADDR(n),                                        \
+		.irq_config_func = mcux_rtwdog_config_func_##n,                                    \
+		.clock_frequency = DT_INST_PROP_BY_PHANDLE(n, clocks, clock_frequency),            \
+		.clk_source = TO_RTWDOG_CLK_SRC(DT_INST_PROP(n, clk_source)),                      \
+		.clk_divider = TO_RTWDOG_CLK_DIV(DT_INST_PROP(n, clk_divider)),                    \
+	};                                                                                         \
+	static void mcux_rtwdog_config_func_##n(const struct device *dev)                          \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), mcux_rtwdog_isr,            \
+			    DEVICE_DT_INST_GET(n), 0);                                             \
+		irq_enable(DT_INST_IRQN(n));                                                       \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(n, &mcux_rtwdog_init, NULL, &mcux_rtwdog_data_##n,                   \
+			      &mcux_rtwdog_config_##n, POST_KERNEL,                                \
+			      CONFIG_KERNEL_INIT_PRIORITY_DEVICE, &mcux_rtwdog_api);
+
+DT_INST_FOREACH_STATUS_OKAY(MCUX_RTWDOG_INIT)
```

Comment:
Missing newline here. You may want to check your editor settings to make sure it adds a newline at the end of the file (some editors can also strip trailing whitespace)